internal/slicer/slicer.go

```diff
@@ -69,13 +71,48 @@ func Run(options *RunOptions) error {
 		targetDirAbs = filepath.Join(dir, targetDir)
 	}
 
+	// Archives are ordered in descending order by priority. The
+	// default priority is 0 and the maximum allowed priority is
+	// the upper limit of int.
+	//
+	// If a package to be downloaded exists in archive A and
+	// archive B, and archive A has higher priority than archive
+	// B, the package is downloaded from archive A even if archive
+	// B has a newer version.
+	//
+	// If two or more archives have the same priority, the most
+	// recent version of the package from all of them is selected,
+	// if any. Otherwise, archives with lower priority are tried,
+	// if any.
+	//
+	// If two or more archives with the same priority contain the
+	// package with the same version, the first archive from the
+	// options.Archives iterator is used. In effect, the choice is
+	// random.
```

Comment:
How about making it deterministic and choosing the archive _(with the highest priority)_ that comes first in the `chisel.yaml` file?

Reply:
Too late for that I'm afraid. If you look two lines below I'm iterating over `options.Archives`, which is already a map. I think this comes from `setup` that reads YAML into a map.

Reply:
As we discussed, while we cannot maintain the order as defined in the YAML (as it's map), we can at least sort the archives by name. See the new _Make package version selection deterministic_ commit:
```
    When two or more archives with highest priority contain a package with
    the same version, the choice is effectively random, because the
    iteration order over options.Archives map is not defined.
    
    Get rid of this nondeterminism by sorting archives by their priority and
    then lexicographically by their labels when the priorities are equal.
    Since archive labels are used as keys in YAML map, they should be unique
    and the order should be total.
```